axelor-production/src/main/java/com/axelor/apps/production/service/SaleOrderLineDummyProductionServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+
+  @Override
+  public Map<String, Object> getOnLoadDummies(SaleOrderLine saleOrderLine, SaleOrder saleOrder)
+      throws AxelorException {
+    Map<String, Object> dummyFields = super.getOnLoadDummies(saleOrderLine, saleOrder);
+    dummyFields.putAll(initProductionInformation(saleOrderLine));
+    return dummyFields;
+  }
+
+  protected Map<String, Object> initProductionInformation(SaleOrderLine saleOrderLine) {
+    Map<String, Object> dummyFields = new HashMap<>();
+
+    List<ManufOrder> manufOrderList = saleOrderLine.getManufOrderList();
+    if (CollectionUtils.isEmpty(manufOrderList)) {
+      return dummyFields;
+    }
+    List<Integer> statusSelectList =
+        manufOrderList.stream().map(ManufOrder::getStatusSelect).collect(Collectors.toList());
+    if (statusSelectList.stream()
+        .allMatch(status -> status == ManufOrderRepository.STATUS_CANCELED)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_CANCELED);
+      return dummyFields;
+    }
+    if (statusSelectList.stream().allMatch(status -> status == ManufOrderRepository.STATUS_DRAFT)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_DRAFT);
+      return dummyFields;
+    }
+    if (statusSelectList.stream()
+        .allMatch(status -> status == ManufOrderRepository.STATUS_STANDBY)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_STANDBY);
+      return dummyFields;
+    }
+    if (statusSelectList.stream()
+        .allMatch(
+            status ->
+                status == ManufOrderRepository.STATUS_FINISHED
+                    || status == ManufOrderRepository.STATUS_CANCELED
+                    || status == ManufOrderRepository.STATUS_MERGED)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_FINISHED);
+      return dummyFields;
+    }
+
+    if (statusSelectList.stream()
+            .anyMatch(status -> status == ManufOrderRepository.STATUS_IN_PROGRESS)
+        || statusSelectList.stream()
+            .anyMatch(status -> status == ManufOrderRepository.STATUS_FINISHED)
+        || statusSelectList.stream()
+            .anyMatch(status -> status == ManufOrderRepository.STATUS_STANDBY)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_IN_PROGRESS);
+      return dummyFields;
+    }
+
+    if (statusSelectList.stream()
+        .anyMatch(status -> status == ManufOrderRepository.STATUS_PLANNED)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_PLANNED);
+      return dummyFields;
+    }
+    if (statusSelectList.stream()
+        .anyMatch(status -> status == ManufOrderRepository.STATUS_STANDBY)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_STANDBY);
+      return dummyFields;
+    }
+    if (statusSelectList.stream().anyMatch(status -> status == ManufOrderRepository.STATUS_DRAFT)) {
+      dummyFields.put(DUMMY_PRODUCTION_STATUS, PRODUCTION_STATUS_DRAFT);
+      return dummyFields;
+    }
+
+    return dummyFields;
+  }
```

Comment:
this service would be easier to read if we just had a protected method returning the correct PRODUCTION_STATUS_*. Also all PRODUCTION_STATUS variable could be defined inside an enum